crowdsourcing/viewsets/user.py

```diff
@@ -155,3 +155,17 @@ def retrieve(self, request, *args, **kwargs):
         user = get_object_or_404(self.queryset, user=request.user)
         serializer = UserPreferencesSerializer(instance=user)
         return Response(serializer.data)
+
+    def update(self, request, user__username=None):
```

Comment:
lookup_field = 'user__username' and lookup_value_regex = '[^/]+'  
and then you can drop user__username